analytics/app/src/main/java/com/google/firebase/quickstart/analytics/java/MainActivity.java

```diff
@@ -104,28 +107,32 @@ protected void onCreate(Bundle savedInstanceState) {
             setUserFavoriteFood(userFavoriteFood);
         }
 
-
         // Create the adapter that will return a fragment for each image.
-        mImagePagerAdapter = new ImagePagerAdapter(getSupportFragmentManager(), IMAGE_INFOS);
+        mImagePagerAdapter = new ImagePagerAdapter(this, IMAGE_INFOS);
 
         // Set up the ViewPager with the pattern adapter.
-        mViewPager = binding.viewPager;
+        mViewPager =  (ViewPager2)findViewById(R.id.viewPager);
         mViewPager.setAdapter(mImagePagerAdapter);
+        mViewPager.setOrientation(ViewPager2.ORIENTATION_HORIZONTAL);
 
-        // Workaround for AppCompat issue not showing ViewPager titles
-        ViewPager.LayoutParams params = (ViewPager.LayoutParams)
-                binding.pagerTabStrip.getLayoutParams();
-        params.isDecor = true;
-
-        // When the visible image changes, send a screen view hit.
-        mViewPager.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
+        mViewPager.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
             @Override
-            public void onPageSelected(int position) {
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+                super.onPageScrolled(position, positionOffset, positionOffsetPixels);
                 recordImageView();
                 recordScreenView();
             }
         });
 
+        TabLayout tabLayout = findViewById(R.id.tab_layout);
```

Comment:
Use [ViewBinding](https://developer.android.com/topic/libraries/view-binding) instead:
```suggestion
        TabLayout tabLayout = binding.tabLayout;
```